### ISSUE TITLE
Having trouble customising block manager

### ISSUE THREAD

**Issue by thecodefish** [artf/grapesjs#107]
I'm trying to customise the editor by adding a number of new block types. Defining the blocks themselves is fine (I've got that working), but I think I need to actually customise (or replace) the block manager itself. When complete we will have far too many blocks to simply list all of them in the side panel - we want to have a mini selection wizard for the user to find the appropriate blocks that they want. This could be a couple of dropdowns to select category/subcategory, etc which then show the few blocks that match the filters. At the end of it, I still want the user to be able to drag their chosen block into the canvas.

Is it possible to customise or replace the block manager in this way? What would you recommend? 

I'm happy to build my own 'block manager' outside of grapeJs (if that's an option) but I need to be able to drag the blocks onto the canvas.

**Comment by artf** [comment]
I think you might be able to do so with changing the block collection and rerendering (not tested)
```js
var blockCollection = editor.BlockManager.getAll();
// We gonna reset the actual collection so we need to keep it somewhere
var clonedCollection = blockCollection.clone();
// For the sake of simplicity, I use where() helper as a filter function
var filtered = clonedCollection.where({label: 'Image'});
// Reset the collection
blockCollection.reset(filtered);
// Rerender
editor.BlockManager.render();
```

**Comment by thecodefish** [comment]
@artf Sorry I typed up a reply to this a few days ago but I must not have posted it. I tried your approach and I managed to get something working that would filter the available blocks for the first render. However if I then tried to change the filter (or even run the same filter again) it would blank out the list. If I debugged things it looked like the collection had the right items in it (eg `editor.BlockManager.getAll()` would have the right number of items) however the UI was rendering nothing on that second pass.

I don't have my exact code on hand unfortunately but here's some rough code for what I was doing:

```
//performed once during init to get a copy of all blocks
var allBlocks = editor.BlockManager.getAll().clone();

//performed on an event (filter changed)
var filteredBlocks = allBlocks.where({label: 'Image'});
editor.BlockManager.getAll().reset(filteredBlocks);
//at this point if I debug editor.BlockManager.getAll() it contains the items I expect
editor.BlockManager.render();
```

Any thoughts on why this code might work the first time, but not the second?
